src/main/webapp/app/shared/components/filter/filter.component.ts

```diff
@@ -0,0 +1,81 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { ActivatedRoute, NavigationExtras, Params, Router } from '@angular/router';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'tiny-filter',
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FilterComponent implements OnInit, OnDestroy {
+
+  @Output() formChanged: EventEmitter<FormGroup> = new EventEmitter<FormGroup>();
```

Comment:
I would call this an anti-pattern to expose the entire form here. Why not implement a custom `ControlValueAccessor` here to encapsulate the behaviour?

Reply:
I wouldn't create a custom form control for the filters, because in this case, beside the filters, I don't have anything else to add in the form. I also wanted to encapsulate the logic for setting/ reading query params in the filters component. What I wanted was to have access, from the parent, on the valueChanges event. I was too lazy to do the unsubscribe manually in the app.component ... , and that is why I went on that direction and used switchMap for canceling the old apis in que. But yes, it would had been better to expose only the valueChanges events, and to unsubscribe previous apis calls manually.

Reply:
Changed the logic in the last commit. Only exposed: filterChanged event from the filter.component.ts